Runtime/Scripts/MobiledgeXSettings.cs

```diff
@@ -1,14 +1,167 @@
 ﻿
 using UnityEngine;
-
+using DistributedMatchEngine;
+using System.Collections.Generic;
+using System.Linq;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
 namespace MobiledgeX
 {
-	 public class MobiledgeXSettings: ScriptableObject
-	{
-	    public string orgName;
-	    public string appName;
-	    public string appVers;
-	    public int TCP_Port;
-	    public int UDP_Port;
-	}
+    // Complier Directive if UNITY_EDITOR is used extensively since MobiledgeXSettings Object is used in production 
+    // Without this complier directive , Unity wont build the app
+#if UNITY_EDITOR
+
+    //using PropertyDrawer interface (used for creating custom views of fields in the inspector)
+    //ReadOnly Attribute makes mapped ports' fields immutable in the inspector
+    public class ReadOnlyAttribute : PropertyAttribute{  }
+
+    [CustomPropertyDrawer(typeof(ReadOnlyAttribute))]
+    public class ReadOnlyDrawer : PropertyDrawer
+    {
+        public override float GetPropertyHeight(SerializedProperty property,
+                                                GUIContent label)
+        {
+            return EditorGUI.GetPropertyHeight(property, label, true);
+        }
+
+        public override void OnGUI(Rect position,
+                                   SerializedProperty property,
+                                   GUIContent label)
+        {
+            GUI.enabled = false;
+            EditorGUI.PropertyField(position, property, label, true);
+            GUI.enabled = true;
+        }
+    }
+#endif
+    // MatchingEngine AppPort is not Serializable
+    // Protocol enum is used to override LProto easier to read
+    [System.Serializable]
```

Comment:
Do we know why AppPort is not serializable? Are we missing a field, or was this the C# json problems that Garner was running into?

Reply:
AppPort have the System.Serializable Attribute on the C# SDK, also LProto is less friendly than Protocol, I can use AppPort later if its updated on the SDK

Reply:
You mean the enum being broken?

Reply:
The enum broken?

Reply:
No its not ,
1. Lproto is less user friendly.
2. LProto.HTTP & LProto.UNKNOWN , while Ports on console are either TCP or UDP
3. AppPort is in the dll so I can add the Attribute System.Serializable to it (Used to to expose a variable in Unity Inspector and to Include Inspector changes with Unity Hot Loading)